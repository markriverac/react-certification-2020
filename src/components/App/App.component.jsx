import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import HomePage from '../../pages/Home';
import NotFound from '../../pages/NotFound';
import FavoritesPage from '../../pages/Favorites';
import Private from '../Private';
import Layout from '../Layout';
import Navigation from '../Navigation/Navigation.component';
import VideoPage from '../../pages/Video';
import UserProvider from '../../providers/UserProvider/User.provider';
import VideoProvider from '../../providers/VideoProvider/Video.provider';

function App() {
  return (
    <UserProvider>
      <BrowserRouter>
        <Navigation />
        <Layout>
          <Switch>
            <VideoProvider>
              <Route exact path="/">
                <HomePage />
              </Route>
              <Route exact path="/video/:videoId">
                <VideoPage />
              </Route>
              <Private exact path="/favorites">
                <FavoritesPage />
              </Private>
            </VideoProvider>
            <Route path="*">
              <NotFound />
            </Route>
          </Switch>
        </Layout>
      </BrowserRouter>
    </UserProvider>
  );
}

export default App;
