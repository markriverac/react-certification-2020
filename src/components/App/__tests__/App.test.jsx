import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../App.component';

Enzyme.configure({ adapter: new Adapter() });

it('Renders without exploding', () => {
  const notFound = renderer.create(<App />);
  const tree = notFound.toJSON();
  expect(tree).toMatchSnapshot();
});
