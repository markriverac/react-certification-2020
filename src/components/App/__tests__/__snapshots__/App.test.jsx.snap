// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without exploding 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark sticky-top"
  >
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <a
      className="navbar-brand"
      href="#home"
    >
      Find Your video
    </a>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </div>
    </div>
    <form
      className="form-inline"
      onSubmit={[Function]}
    >
      <input
        aria-label="Search"
        className="form-control mr-sm-2 login-input"
        onChange={[Function]}
        placeholder="username"
        type="search"
        value=""
      />
      <button
        className="btn btn-outline-secondary my-2 my-sm-0"
        type="submit"
      >
        LogIn
      </button>
    </form>
  </nav>,
  <div
    className="page-container"
  >
    <main
      className="container"
    >
      <section
        className="homepage"
      >
        <div
          className="ui container"
        >
          <div>
            <form
              className="ui form"
              onSubmit={[Function]}
            >
              <div
                className="field"
              >
                <input
                  className="main-search"
                  htmlFor="searchBar"
                  name="video-search"
                  onChange={[Function]}
                  placeholder="Tell me what you need"
                  type="text"
                  value=""
                />
              </div>
            </form>
          </div>
          <div>
            <ul
              className="list-unstyled"
            />
          </div>
        </div>
      </section>
    </main>
  </div>,
]
`;
